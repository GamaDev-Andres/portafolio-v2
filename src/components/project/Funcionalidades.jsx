import PropTypes from 'prop-types'

const Funcionalidades = ({ proyecto }) => {
  const formatId = (idNoFormat) => {
    return idNoFormat.split(' ').join('')
  }
  return (
    <div>
      <h2 className="font-bold text-letra dark:text-letraDark text-lg">
        Funcionalidades
      </h2>
      <ul className="pl-4 inline-block">
        {proyecto.detalles.map((el) => (
          <li className="list-disc hover:text-blue-400 " key={el.titulo}>
            <a className="underline" href={`#${formatId(el.titulo)}`}>
              {el.titulo}
            </a>
          </li>
        ))}
      </ul>
    </div>
  )
}

Funcionalidades.propTypes = {
  proyecto: PropTypes.object.isRequired
}
export default Funcionalidades
