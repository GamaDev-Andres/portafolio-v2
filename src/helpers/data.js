export const proyectos = [
  {
    titulo: 'BARBERAPP',
    descripcion: `Aplicación realizada con el stack MERN y styled components,
     enfocada en la administración de una barbería en donde el cliente tiene 
     permitido agendar una cita con un barbero basándose en su perfil ,
      cortes o calificación de otros clientes.`,
    tecnologias: [
      'React',
      'Express',
      'StyledComponents',
      'MongoDb',
      'Reducers'
    ],
    url: 'https://barberappag.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/BarberApp-fe',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644874961/portafolio/barberapp/barberapp_sdpgwl.png',
    fotos: []
  },
  {
    titulo: 'NETBOOK',
    descripcion:
      'Mini clon de una red social realizado con React, Redux y firebase, me baso en ciertas funcionalidades de Facebook, tales como, publicaciones, likes, comentarios y perfiles individuales.',
    tecnologias: ['React', 'Firebase', 'Redux', 'Sass'],
    url: 'https://agnetbook.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/network-social-react',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644874998/portafolio/netbook/netbook_gus84s.png',
    fotos: []
  },
  {
    titulo: 'GALERIA',
    descripcion:
      'Simulación de una galería web realizada con React y Firebase , la aplicación se basa en la posibilidad de crear álbumes y tomarse fotos directamente en el álbum deseado y también tiene la posibilidad de buscar imágenes en una API y guardarlas en un álbum.',
    tecnologias: ['React', 'Firebase', 'StyledComponents'],
    url: 'https://practica-galeria.web.app/',
    codigo: 'https://github.com/GamaDev-Andres/react-galeria',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644875042/portafolio/galeriareact/reactFirebase_xjiauw.png',
    fotos: []
  },
  {
    titulo: 'MIS PELIS',
    descripcion:
      'Proyecto de películas hecho con Javascript, Sass y webpack .En este proyecto me baso en un diseño de una pagina funcional, consulto la API IMDB que me provee información tanto de series como de películas con sus respectivos tráileres y sinopsis. Implemento buscadores y lista de favoritos.',
    tecnologias: ['Html', 'Sass', 'Webpack', 'Javascript'],
    url: 'https://afmispelisgi.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/PeliculasOMDB',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644875064/portafolio/mispelis/imagepelisOMDB_cchblf.png',
    fotos: []
  },
  {
    titulo: 'FINDYOURGAMES',
    descripcion:
      'Proyecto realizado con Html, css, javascript y gulp. En este proyecto consulto una API de videojuegos y organizo su información en una interfaz que me permite hacer búsqueda y filtrado de los videojuegos disponibles.',
    tecnologias: ['Html', 'Sass', 'Gulp', 'Javascript'],
    url: 'https://findyourgames.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/FindYourGames',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644875081/portafolio/findyourgames/proyecto1_vsncjy.png',
    fotos: []
  }
]

export const miDescripcion = {
  quienSoy: {
    parrafo1: `Mi nombre es Andres Gama, me apasiona encontrar grandes retos que me
          permitan crecer profesional y personalmente, la programación me
          permite enfocarme en esto.Actualmente tengo 21 años y sé que lograre
          que mis conocimientos sean participes de grandes, útiles y reconocidos
          proyectos.`,
    parrafo2: `En mí podras ver a un chico ansioso por aprender y dispuesto a
          afrontar complejas curvas de aprendizaje y gracias a mi constancia y
          disciplina veras a alguien creciendo constantemente.`
  },
  proceso: {
    parrafo1: `Inicie con pequeños cursos y blogs de programación en YouTube,
            posteriormente me di cuenta de que este mundo de verdad me
            interesaba y a pesar de que lo miraba complicado ... debía
            intentarlo entonces me aventuré a profundizar y empecé a conocer
            plataformas como Udemy, Platzi, MDN, etc...`,
    parrafo2: `He tenido diferentes dificultades en mi proceso, tanto para aprender
            como para aplicar lo aprendido, entre estas dificultades se destaca
            el hecho de enfrentarme a retos desconocidos como pueden ser
            tecnologias nuevas o profundizar más una tecnología para obtener el
            resultado deseado o uno más optimo, al final entendí que lo más
            lindo de esta profesión es que se garantiza un crecimiento
            constante.`
  },
  presentacion:
    'apasionado por el mundo del desarrollo y por crear aplicaciones e interfaces que aporten y brinden grandes experiencias.'
}

export const links = {
  contacto: [
    {
      href: 'https://drive.google.com/file/d/1IIUutqCGZL_KSN7NqE-JdRYQl_6Qaqhd/view',
      type: 'Resumé'
    }
  ]
}
