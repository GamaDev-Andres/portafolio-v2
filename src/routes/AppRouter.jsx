import { useState } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from '../components/home/Home';
import Project from '../components/project/Project';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/project/:id" element={<Project />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;
