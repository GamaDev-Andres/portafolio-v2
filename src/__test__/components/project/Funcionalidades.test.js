/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { HelmetProvider } from 'react-helmet-async'
import { MemoryRouter } from 'react-router-dom'
import Funcionalidades from '../../../components/project/Funcionalidades'
import { proyectos } from '../../../helpers/data'
import AppRouter from '../../../routes/AppRouter'
describe('probando <Funcionalidades/>', () => {
  const proyecto = proyectos.find(project => project.detalles)
  test('debe montarse correctamente', () => {
    const wrapper = Enzyme.mount(
      <Funcionalidades proyecto={proyecto} />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('h2').text().trim()).toBe('Funcionalidades')
  })
  test('debe montarse correctamente en ruta "project" ', () => {
    const wrapper = Enzyme.mount(
      <MemoryRouter initialEntries={['/project/BARBERAPP']}>
        <HelmetProvider>
          <AppRouter />
        </HelmetProvider>
      </MemoryRouter>
    )
    expect(wrapper.find('Funcionalidades').exists()).toBe(true)
  })
})
