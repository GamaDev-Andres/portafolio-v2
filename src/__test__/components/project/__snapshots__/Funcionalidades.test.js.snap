// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <Funcionalidades/> debe montarse correctamente 1`] = `
<div>
  <h2
    className="font-bold text-letra dark:text-letraDark text-lg"
  >
    Funcionalidades
  </h2>
  <ul
    className="pl-4 inline-block"
  >
    <li
      className="list-disc hover:text-blue-400 "
      key="Video presentacion de InstaGama"
    >
      <a
        className="underline"
        href="#VideopresentaciondeInstaGama"
      >
        Video presentacion de InstaGama
      </a>
    </li>
  </ul>
</div>
`;
